packages/std/src/errors/system_error.rs

```diff
@@ -0,0 +1,36 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::HumanAddr;
+
+/// SystemError is used for errors inside the VM and is API frindly (i.e. serializable).
+///
+/// This is used on return values for Querier as a nested result: Result<StdResult<T>, SystemError>
+/// The first wrap (SystemError) will trigger if the contract address doesn't exist,
+/// the QueryRequest is malformated, etc. The second wrap will be an error message from
+/// the contract itself.
+///
+/// Such errors are only created by the VM. The error type is defined in the standard library, to ensure
+/// the contract understands the error format without creating a dependency on cosmwasm-vm.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
```

Comment:
What about the other ones from `StdError`:

```rust
#[serde(rename_all = "snake_case")]
#[non_exhaustive]
```

Maybe they belong here as well? At least `snake_case`

Reply:
Probably yes. I did not change this code, just moved it around. I'll have a look.

Reply:
`snake_case` is now set after rebasing on master, where you set it.

I leave non_exhaustive up to you, now that you know what it is.

Reply:
I'd add `#[non_exhaustive]` for now, it makes changes less breaking in the public API, and we can remove these when we hit 1.0 and finalize the error cases.

Reply:
Done in https://github.com/CosmWasm/cosmwasm/pull/305/commits/462e889cc902b92d52a3c284a4fc8405181ef6f1